lib/msal-angular/src/msal.service.ts

```diff
@@ -4,204 +4,78 @@
  */
 
 import { Inject, Injectable } from "@angular/core";
+import { Location } from "@angular/common";
 import {
-    UserAgentApplication,
-    Configuration,
-    AuthenticationParameters,
-    AuthResponse,
-    AuthError,
-    authResponseCallback,
-    errorReceivedCallback,
-    tokenReceivedCallback,
-    UrlUtils
-} from "msal";
-import { Router } from "@angular/router";
-import {BroadcastService} from "./broadcast.service";
-import { MSALError } from "./MSALError";
-import { MsalAngularConfiguration } from "./msal-angular.configuration";
-import { MSAL_CONFIG, MSAL_CONFIG_ANGULAR } from "./constants";
-
-import { Minimatch } from "minimatch";
-
-const buildMsalConfig = (config: Configuration) : Configuration => {
-    return {
-        ...config,
-        framework: {
-            ...config.framework,
-            isAngular: true
-        }
-    };
-};
+    IPublicClientApplication,
+    AccountInfo,
+    EndSessionRequest,
+    AuthorizationUrlRequest,
+    AuthenticationResult,
+    PopupRequest,
+    RedirectRequest,
+    SilentRequest
+} from "@azure/msal-browser";
+import { MSAL_INSTANCE } from "./constants";
+import { Observable, from } from "rxjs";
+
+export interface IMsalService {
+    acquireTokenPopup(request: PopupRequest): Observable<AuthenticationResult>;
+    acquireTokenRedirect(request: RedirectRequest): Observable<void>;
+    acquireTokenSilent(silentRequest: SilentRequest): Observable<AuthenticationResult>;
+    getAccountByUsername(userName: string): AccountInfo | null;
```

Comment:
We should add `getAccountByHomeId` here because `getAccountByUsername` won't work for B2C without additional setup

Reply:
+`getAccountByLocalId()`

Reply:
And probably worth it to move to its own file.

Reply:
And this actually brings up an interesting question: should `IMsalService` only include properties that are being turned into Observables?

Reply:
I would say yes, as that's the only reason why it exists. Apps can still get at the original msal instance for the non-async properties.

Reply:
Thanks for the feedback! We made the instance of msal public so that `getAccountByHomeId`, `getAccountByUsername` and `getAccountByLocalId` can be accessed from that instance instead of from the service. `IMsalService` is now in its own file, and only has properties that are being turned into Observables.